types/index.d.ts

```diff
@@ -14,7 +14,7 @@ declare namespace inject {
 
   export type DispatchFunc = http.RequestListener
 
-  export type CallbackFunc = (err: Error, response: Response) => void
+  export type CallbackFunc = (err: Error | undefined, response: Response | undefined) => void
```

Comment:
Imho this is better:

```suggestion
  export interface CallbackFunc {
    (err: Error, response: never): void;
    (err: undefined | null, response: Response): void;
  }
```

https://www.typescriptlang.org/play?ssl=5&ssc=1&pln=1&pc=1#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgYUwBsiAjHAawDEBXJbOAbwFgAoOOACjSgC44AolCjQANHCjAAzpCRTg-JMABuaAJT9lEBABMA3Gw7dh-OjuDpkwHXAA+cJDRLjJMiHIVwAStNnyNcFq6BqwAvmxs2O5S8Nik-IQk5NjUdAwAvHDoaTAI7lw8Lr7RwGpMhojoBcJlLOwccAD0jXAAFphIOkR4PNAVHFFy8L1QcJk8-XAjAHQAttJSmADmwJOufsDTmMKYAJ4AQjTo6GiTzXDTlxWhU0Ty5fUcPHMLy3jnI2ct7Z3dEsUeNYA+RbHYHI4nKBfC5XerhMJ6IA

Reply:
@types/node use
```js
(err: Error | null, ...) => void
```

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/dns.d.ts#L118

Reply:
Anyway, it is the same
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgYUwBsiAjHAawDEBXJbOAbwFgAoOOACjSgC44AolCjQANHCjAAzpCRTg-OgBNg6ZMCUBKfgDcICJQG42HbsMVIVapBrgAfOEholxkmRDkK4AJWmz52nB6BsasAL5sbKCQsHAwAJ5geAIAjjTEADwAGuIAmgB8cAC8JlwZACr5nJrFheVwoKiWUnBZcAD8cACMcPwATDWNwM2lnBVVNUV1DSBNSi25Hd29cANLMFA0ePxYRPKhbAlJcJjFgmmZhCTk2NR02OJmfILC0PZwyqrqSu9OLhJ+Dzyfi+dyed6fawaSaFYJKfJAA

Reply:
Anyway, it is the same
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgYUwBsiAjHAawDEBXJbOAbwFgAoOOACjSgC44AolCjQANHCjAAzpCRTg-OgBNg6ZMCUBKfgDcICJQG42HbsMVIVapBrgAfOEholxkmRDkK4AJWmz52nB6BsasAL5sbKCQsHAwAJ5geALgwNgwADwAKnCgqJZScVA0wAB8cAC8cFlR4NDwCUmCAI40xBkAGuIAmuUVJlzZpZyaleU5ecAFcB1wAPxwAIxw-ABMo5MFA5xDI2PVuSD5SoXd80srcOvnMMV4-FhE8qFs2B5S8Nik-IQk5NjUOgMKroIEwBAeLg8Vx+d7AUYsdiIdBQ4QIgYcRp4TCVQSpdIZAStdpYiAo6GCYTQUqlDFxRJ4Ui4lJJAlEtpEDKklFufzAcS+dyeexwZSqdRKGkDMK5J54REcTEMuA4qostKZdkkhlk3LCcRi6y2BxOEhSpFKppMtX4zXEznciSwzwC53yEWGiXmjgRcKhIA

Reply:
What is your claim?

Reply:
I see the sense of making it simpler and conforming to the general style as an example from @types/node

Reply:
Does anyone else have any thoughts on this?

maybe @mcollina @Eomm

Reply:
follow whatever the official `@types/node` are implemented